tls/s2n_psk.c

```diff
@@ -342,3 +354,13 @@ int s2n_connection_set_external_psks(struct s2n_connection *conn, struct s2n_ext
 
     return S2N_SUCCESS;
 }
+
+int s2n_config_choose_psk_cb(struct s2n_connection *conn, s2n_psk_selection_callback cb)
```

Comment:
Nit on the naming: We probably want to match the type. Also I think most APIs like this include a `set` in the name. I also think there's more that use `_callback` rather than `_cb`:

```suggestion
int s2n_config_set_psk_selection_callback(struct s2n_connection *conn, s2n_psk_selection_callback cb)
```

Reply:
You also might want this method to live in s2n_config.c, instead of s2n_psk.c.